### PR TITLE
287-format pod container port name

### PR DESCRIPTION
**Pull request by yangcao77**
### What does this PR do?
This PR formats the container port name to be `<portNumber>-<protocol>`, updated to use latest api repo and latest schema.

### What issues does this PR fix or reference?
https://github.com/devfile/api/issues/287

### Is your PR tested? Consider putting some instruction how to test your changes
unit tested

### CODE REVIEW

**Comment by yangcao77** [comment]
@mmulholl Some of the generated random strings in api tests are not valid. I have to modify them to match the schema. Please help review the test code change, thanks!

**Comment by mmulholla** [comment]
@yangcao77 Hi, the changes you made to the parser tests look good.

File: pkg/devfile/generator/utils.go [hunk 1]
```diff
@@ -34,13 +33,25 @@ func convertEnvs(vars []v1.EnvVar) []corev1.EnvVar {
...
 
+		if endpoint.Protocol == v1.UDPEndpointProtocol {
+			portProtocol = corev1.ProtocolUDP
+		} else {
+			portProtocol = corev1.ProtocolTCP
+		}
+		name := fmt.Sprintf("%d-%s", portNumber, strings.ToLower(string(portProtocol)))
+		if _, exist := portMap[name]; exist {
+			continue
+		}
```
Comment (maysunfaisal) [inline pkg/devfile/generator/utils.go:49]:
```suggestion
		if _, exist := portMap[name]; !exist {
	                  portMap[name] = true
	                  containerPorts = append(containerPorts, corev1.ContainerPort{
		                  Name:          name,
		                  ContainerPort: portNumber,
		                  Protocol:      portProtocol,
	                  })
                  }
```

would probably do this way but upto you

File: pkg/devfile/generator/utils_test.go [hunk 2]
```diff
@@ -127,12 +128,14 @@ func TestConvertPorts(t *testing.T) {
 			},
 			want: []corev1.ContainerPort{
 				{
-					Name:          endpointsNames[0],
+					Name:          "8080-tcp",
```
Comment (maysunfaisal) [inline pkg/devfile/generator/utils_test.go:131]: Just noticed, can you update teh test case name, it seems wrong

**Review by maysunfaisal — COMMENTED** [review]
suggestion for changing package alias to be less confusing between all these v1 and v2

File: pkg/devfile/generator/utils.go (hunk 1 above)
Comment (yangcao77) [reply pkg/devfile/generator/utils.go]: updated

File: pkg/devfile/generator/utils_test.go (hunk 2 above)
Comment (yangcao77) [reply pkg/devfile/generator/utils_test.go:131]: updated

**Review by maysunfaisal — APPROVED** [review]
/lgtm

**Pull request merged**